js/src/offcanvas.js

```diff
@@ -174,15 +174,28 @@ class Offcanvas extends BaseComponent {
       this.hide()
     }
 
+    const keydownCallback = event => {
+      if (!this._config.keyboard) {
+        EventHandler.trigger(this._element, EVENT_HIDE_PREVENTED)
+        return
+      }
+
+      if (event.key === ESCAPE_KEY) {
+        this.hide()
+      }
+    }
+
     // 'static' option will be translated to true, and booleans will keep their value
     const isVisible = Boolean(this._config.backdrop)
+    const keydownEnabled = Boolean(this._config.keyboard)
```

Comment:
```suggestion
    const keydownEnabled = this._config.keyboard
```

it works only as boolean, so it would be ok just to do a simple check